src/Components/Facility/HospitalList.tsx

```diff
@@ -35,6 +35,17 @@ export const HospitalList = () => {
   } = useFilters({
     limit: 14,
   });
+
+  useEffect(() => {
+    if (Object.keys(qParams).length === 0) return;
+    if (!qParams.state) {
+      advancedFilter.removeFilters(["district", "local_body"]);
+    }
+    if (!qParams.district) {
+      advancedFilter.removeFilters(["local_body"]);
+    }
+  }, [advancedFilter, qParams]);
```

Comment:
I still find `if (Object.keys(qParams).length === 0) return;` line hard to understand why it's there. Maybe would the following be a more readable and correct approach of what you are trying to do? 

cc: @Ashesh3 

```suggestion
    if (!qParams.state && (qParams.district || qParams.local_body)) {
      advancedFilter.removeFilters(["district", "local_body"]);
    }
    if (!qParams.district && qParams.local_body) {
      advancedFilter.removeFilters(["local_body"]);
    }
  }, [advancedFilter, qParams]);
```

Reply:
The suggestion seems good and reads better.
- If state is not present, but district or local_body is, then remove both
- If district is not preset but we still have local_body, then remove it.